Rewrite build file to be strongly typed

JLLeitschuh opened the pull request:
This is built on top of #69

You can take it or leave this PR.
I just wanted to see how hard the exercise of rewriting the build to use the Gradle Kotlin DSL would be. Evidently it wasn't that hard. This took me ~25 min to do.

JLLeitschuh commented on build.gradle.kts:
```diff
@@ -0,0 +1,225 @@
+    classifier = "sources"
+}
+
+tasks.withType<Jar> {
```
There used to be a `classifier=null` here when this was done using the shadow plugin. I don't know why that was there. I don't think it was strictly necessary.

SamCarlberg replied:
Probably to remove the `-all` or `-shadow` suffix from the file name

JLLeitschuh replied:
Fixed, see line 150

JLLeitschuh commented on build.gradle.kts:
```diff
@@ -0,0 +1,225 @@
+val Project.`java`: org.gradle.api.plugins.JavaPluginConvention get() =
+convention.getPluginByName<org.gradle.api.plugins.JavaPluginConvention>("java")
+
+
```
Extra white space I should remove

nightpool commented on build.gradle.kts:
```diff
@@ -0,0 +1,221 @@
+/**
+ * Configures the [checkstyle][org.gradle.api.plugins.quality.CheckstyleExtension] project extension.
+ */
+fun Project.`checkstyle`(configure: org.gradle.api.plugins.quality.CheckstyleExtension.() -> Unit): Unit =
```
Is the fully qualified package name necessary?

JLLeitschuh replied:
This is what you get printed when you run the `gskGenAccessors` task. I just copied and pasted them. 

There's a different mode you can use that doesn't require these declarations in this file but I've never tried it. I could try it and see if it works.

nightpool commented on src/main/java/edu/wpi/first/shuffleboard/widget/NetworkTableTreeWidget.java:
```diff
@@ -38,7 +39,7 @@
    */
   private static final Comparator<TreeItem<NetworkTableEntry>> alphabetical
-      = Comparator.comparing(item -> item.getValue().getKey().toLowerCase());
+      = Comparator.comparing(item -> item.getValue().getKey().toLowerCase(Locale.getDefault()));
```
unrelated change

JLLeitschuh commented on gradle.properties:
```diff
@@ -0,0 +1 @@
+org.gradle.script.lang.kotlin.accessors.auto=true
```
This allowed me to not have all that extra fully classified cruft at the bottom of the `build.gradle.kts` file. Thanks @nightpool for suggesting that I clean that up or I wouldn't have tried this new config mode.

JLLeitschuh commented:
I cleaned this PR up quite a bit this morning. I re-added the shadow plugin. I tested the shadow JAR that the build produces and it runs shuffleboard.

I also added configuring the jar manifest in this PR so now the shadow jar will actually execute the main class when you run `java -jar shuffleboard.jar`. I don't know if this worked before but it definitely works now.
